GenericsHomework/Node.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+namespace GenericsHomework;
+
+
+public class Node<T>
+{
+
+    public Node<T> Next { get; private set; }
+    public T Data { get; }
+
+    public Node(T data)
+    {
+        Data = data;
+        Next = this;
+    }
+
+    public void Append(T value)
+    {
+        if(Exists(value))
+        {
+            throw new InvalidOperationException(nameof(value));
+        }   
+        Node<T> newNode = new(value);
+        Node<T> cur = this;
+
+        while (cur.Next != this)
+        {
+            cur = cur.Next;
+        }
+
+        newNode.Next = cur.Next;
+        cur.Next = newNode;
+    }
+
+    public override string ToString()
+    {
+        Node<T> cur = this;
+        string result = "";
+        while (cur.Next != this)
+        {
+            result += cur.Data + " -> ";
+            cur = cur.Next;
+        }
+        result += cur.Data;
+        return result;
```

Comment:
```suggestion
        if (Value == null) return null;
        return Value.ToString();
```
I think this can be simplified. When I to string a node, I don't think I want to get the whole "list"